sdk/identity/Azure.Identity/src/ManagedIdentityClient.cs

```diff
@@ -0,0 +1,419 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Identity
+{
+    internal class ManagedIdentityClient
+    {
+        private const string MsiEndpointEnvironemntVariable = "MSI_ENDPOINT";
+        private const string MsiSecretEnvironemntVariable = "MSI_SECRET";
+        private static Lazy<ManagedIdentityClient> s_sharedClient = new Lazy<ManagedIdentityClient>(() => new ManagedIdentityClient());
```

Comment:
Per our offline conversation we don't need to be able to set api-version in the ctor since this client is internal and not constructible by other libraries.

---
In reply to: [297384112](https://github.com/Azure/azure-sdk-for-net/pull/6699#discussion_r297384112) [](ancestors = 297384112)